### PR TITLE
Enable creation of compose artifacts on Jitpack

### PR DESCRIPTION
**Pull request by LachlanMcKee**
**Description**:
Enable creation of compose artifacts on Jitpack. These artifacts are named differently as they are build using different dependencies.

**Related issue(s)**:

### CODE REVIEW

File: jitpack.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+jdk:
+  - openjdk8
+install:
+  - echo "Building non-compose artifacts"
+  - ./gradlew clean install -xtest
+  - cp -R libraries libraries-non-compose
```
Comment (LachlanMcKee) [inline jitpack.yml:6]: By copying this to a different directory, we avoid deleting it when we do a gradle clean, but thankfully Jitpack still finds it!

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+// Without this, the plugins module will not be able to use this script without hacks.
+File rootDirectory = rootDir
+while (!rootDirectory.listFiles().any { it.isDirectory() && it.name == ".github" }) {
+    rootDirectory = rootDirectory.parentFile
+}
+
+apply from: "$rootDirectory/gradle/compose.gradle"
+
+def kotlinVersion
+if (useCompose.toBoolean()) {
```
Comment (CherryPerry) [inline gradle/dependencies.gradle:11]: We are free to set `1.4.30` for both regular and compose variants.

File: jitpack.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+jdk:
+  - openjdk8
+install:
+  - echo "Building non-compose artifacts"
+  - ./gradlew clean install -xtest
```
Comment (CherryPerry) [inline jitpack.yml:5]: Is not `publishToMavenLocal` enough for JitPack?

https://jitpack.io/docs/BUILDING

Comment (LachlanMcKee) [reply jitpack.yml:5]: Hmm. this was the recommendation from Jitpack when using a custom build script:
https://jitpack.io/docs/BUILDING/#custom-commands

File: gradle/dependencies.gradle (hunk 2 above)
Comment (LachlanMcKee) [reply gradle/dependencies.gradle:11]: we certainly can, but perhaps that should be in a different ticket as there might be unexpected changes as a result of this?

At least with the `useCompose` flag, we isolate any potential issues away from the 'stable' binary. Happy to fast follow this PR with an update Kotlin PR. If precede this PR with a Kotlin upgrade.

File: jitpack.yml (hunk 3 above)
Comment (CherryPerry) [reply jitpack.yml:5]: Does our Gradle config even has `install` command? Can you try to run it?

Comment (LachlanMcKee) [reply jitpack.yml:5]: it does. I have successfully run this script on a fork.
https://jitpack.io/#LachlanMcKee/RIBs/0.27.2.2

Comment (LachlanMcKee) [reply jitpack.yml:5]: image

File: sandbox/src/test/java/com/badoo/ribs/sandbox/rib/switcher/SwitcherInteractorTest.kt [hunk 4]
```diff
@@ -21,7 +21,10 @@ import com.nhaarman.mockitokotlin2.verify
 import io.reactivex.Observable
 import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
 
+@RunWith(RobolectricTestRunner::class)
```
Comment (LachlanMcKee) [inline sandbox/src/test/java/com/badoo/ribs/sandbox/rib/switcher/SwitcherInteractorTest.kt:27]: When adding the `deps.compose.ui` dependency, the unit tests fail with the following error:
https://github.com/badoo/RIBs/runs/1981140159?check_suite_focus=true

`java.lang.RuntimeException: Method getMainLooper in android.os.Looper not mocked.`

Comment (LachlanMcKee) [reply sandbox/src/test/java/com/badoo/ribs/sandbox/rib/switcher/SwitcherInteractorTest.kt:27]: Running this with Robolectric appears to resolve the issue. I presume the dependency is causing mischief

File: sandbox/src/test/java/com/badoo/ribs/sandbox/rib/menu/MenuRibTest.kt [hunk 5]
```diff
@@ -23,6 +25,7 @@ import org.junit.Test
  * assertions based on view model emissions
  * It runs a lot faster than UI test as it doesn't require real device or emulator
  */
+@RunWith(RobolectricTestRunner::class)
```
Comment (LachlanMcKee) [inline sandbox/src/test/java/com/badoo/ribs/sandbox/rib/menu/MenuRibTest.kt:28]: Same as the other comment below

**Review by CherryPerry — APPROVED** [review]

**Pull request merged**